hack/make/run

```diff
@@ -55,6 +59,29 @@ args="--debug \
 	$storage_params \
 	$extra_params"
 
+if [ -n "$DELVE_PORT" ]; then
+	dockerd="dlv --listen=0.0.0.0:$delve_listen_port \
+		--headless=true \
+		--log \
+		--api-version=2 \
+		--only-same-user=false \
+		--check-go-version=false \
+		--accept-multiclient \
+		exec ${dockerd} --"
+fi
+
 echo "${dockerd} ${args}"
-# shellcheck disable=SC2086
-exec "${dockerd}" ${args}
+
+if [ -n "$DOCKER_ROOTLESS" ]; then
+	# shellcheck disable=SC2068
+	# shellcheck disable=SC2086
+	exec sudo -u $user \
+		-E DOCKERD="$dockerd" \
+		-E XDG_RUNTIME_DIR=/tmp/docker-${uid} \
+		-E XDG_CONFIG_HOME=/home/${user}/.config \
+		-E HOME=/home/${user} \
+		-- /go/src/github.com/docker/docker/contrib/dockerd-rootless.sh ${args}
```

Comment:
This is a bit unfortunate, but as this is all within the container 🤷‍♂️ 

Was thinking if there's other solutions (for a follow-up);

- make `make install` also install this script into a standard location
- but ... for the dev container, I was actually thinking of adding the `bundles/...` location to `PATH`, so that `make install` is not needed after rebuilding the daemon
- (perhaps make `hack/make.sh binary` also copy the script to `bundles/..` ? 🤔 


Anyway; more for a follow-up (suggestions welcome!)

Reply:
:+1 for less steps, better inner loop
maybe `hack/make.sh dev` could build, install and run with debugger enabled?

Reply:
`hack/make.sh dev` would be run inside the container, so it would "work", but requires the `make shell` to be started with the delve port published.

We _could_ consider mapping the delve port by default (on localhost), as `make shell` / `make dev` is meant for local development (mapping the port wouldn't do anything if delve is not running, other then mapping the port)

Reply:
Sounds great, personally I find it a bit cumbersome to type in 3 separate commands to develop the daemon. :joy: Especially since the bash history in the container is not persisted between its recreations, meaning that the reverse search cannot be used to quickly look up the previous commands.

Reply:
In either case, I think those are fine to be done in a follow-up (if we don't want it to block this PR)

Reply:
yeah, it can be a bit cumbersome; I usually have two shells open into the container; one to run the daemon (in which I `hack/make.sh binary && make install && dockerd --debug`), and one to run  other commands, but making that easier would definitely be 👍

Reply:
I'd say that could be something for a separate PR. Not because it's blocking the issue from getting merged, but rather since it's starting to introduce new, potentially unrelated changes - `./hack/make.sh dev` could as well build, install and run the daemon without the debugger and still be handy.

Also, it leads to separate discussion - should Delve wait for the debugger to be attached before running the daemon or start it and allow developer to attach whenever they want? The former could quickly get annoying, but the latter gives little time for the developer to attach the debugger if they want to analyze the daemon's startup.

Reply:
Agreed; definitely not for this PR, but for a follow-up